src/main/java/com/example/springbootrestexample/configurations/MemoryConfiguration.java

```diff
@@ -0,0 +1,13 @@
+package com.example.springbootrestexample.configurations;
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class MemoryConfiguration implements Condition {
```

Comment:
Nice implementation but it could be done better with profiles. Check previous comment.